### DISCUSSION TITLE
Quotes and Quote-to-Order flow

### DISCUSSION THREAD

**Discussion by TLeszczynskiWA** [medusajs/medusa#4748] [Q&A]
Hey there,

I'm new to MedusaJS community, hopefully the following question won't be that silly! 

I wonder if MedusaJS supports Quote management and offers Quote-to-Order flow implementation in some shape or form? I couldn't find anything in docs, neither do I see any entity for Quotes.

**Comment by srindom** [comment]
Depending on the requirements for the business workflow, you may get the functionality you are looking for through the [Draft Order API](https://docs.medusajs.com/modules/orders/draft-orders).

The Draft Order API has the following features:
- Admins can create draft orders, add products, add "ad-hoc" items, set custom prices, shipping methods, discounts, etc.
- Once the draft order is created, the Admin can send an email to the customer.
- Email can be configured to contain a "View and confirm" link.
- Customer can view and pay the Draft Order which will transform it to a regular Order.

Behind the scenes a Draft Order is really just a Cart - this means that the Customer facing flow is just interacting with the Cart API, e.g. for Payment Session management and Cart completion.

One thing to note here is that you must implement the Customer-facing flow. I.e., you should create a frontend that can display the draft order collect payment details from the customer, and call the Complete Cart endpoint. 

Curious to hear if this covers your needs?

**Reply by TLeszczynskiWA** [reply]
Thanks for the reply @srindom ! I could probably make the above flow work in our scenario. However, I'm curious if the Quote flow in its proper form (that is, Customer made Quote that is later transformed into an Order) is something that is planned to be implemented in MedusaJS? An example of a Quoute-to-Order flow I had in mind could be [CommerceTool's implementation](https://docs.commercetools.com/api/quotes-overview#intended-workflow). I hope it's not rude to point you to their docs, if so I apologise!

This is primarily useful for service-based businesses where sale's parameters (cost etc.) need to be assessed individually based on provided information (i.e. location, appartment’s living area and availability of workers for cleaning services). Having said that, it can also be useful in more specific ecommerce scenarios (i.e. a particularly big order) where the customer says "I want those products" (creates the cart), adds a comment saying "I want 25% off" or something similar and then a dedicated department picks it up and either rejects or enters a sort of automated "haggling" process, meaning the company makes an offer for the cart, customer accepts or sends back an edit request and so on.

**Reply by maxoghana** [reply]
Hi, I'm also interested  for the request for quotation process to be implemented in Medusa.

In my use case, we are a furniture rental company. We first receive a RFQ from our customer and we can check availability of the products, calculate the shipping cost and other services cost.

We need something basic : our customer add the products and services to the cart. Provide more information during checkout like rental period, how to access to the place ( ground floor, elevator, etc). We and the customer receive an email with the RFQ details. 

Then, end of the process in medusa.
we send the quotation using another software.

**Reply by radoslaw-sz** [reply]
hi @maxoghana  - you may want to check this: https://github.com/medusajs/b2b-starter-medusa/. It is a starter which includes also quotas flow

**Reply by maxoghana** [reply]
@radoslaw-sz  : thank you. This starter will be very helpfull. I can customise the quote module.to get exactly what I need
